### PR TITLE
[Misc] Cleanup duplicated flags.

### PR DESCRIPTION
**Pull request by frozenbugs**
## Description

Note that the priority of flag setting is config.cmake > -DOPTION=VALUE > dgl_option, which will cause -DOPTION can not working, so we decided to remove config.cmake mechanism in DGL build.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

**Comment by frozenbugs** [comment]
Partially resolve: https://github.com/dmlc/dgl/issues/5475

File: cmake/config.cmake [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
-# Misc.
-#---------------------------------------------
-# Whether to build cpp unittest executables.
-set(BUILD_CPP_TEST OFF)
-
-# Whether to enable OpenMP.
-set(USE_OPENMP ON)
-
-# Whether to build PyTorch plugins.
-set(BUILD_TORCH ON)
```
Comment (Rhett-Ying) [inline cmake/config.cmake:44]: in `CMakeLists.txt`, it's `dgl_option(BUILD_TORCH "Build the PyTorch plugin" OFF)`. So before this change, `BUILD_TORCH` is always on?

Comment (frozenbugs) [reply cmake/config.cmake:44]: No, this cmake/config.cmake was never included.

Comment (frozenbugs) [reply cmake/config.cmake:44]: The included config.cmake were DGL_HOME/config.cmake and DGL_HOME/build/config.cmake.

**Review by Rhett-Ying — APPROVED** [review]

**Pull request merged**